test/TickMath.starknet.ts

```diff
@@ -0,0 +1,188 @@
+      const min = await tickMath.getSqrtRatioAtTick_986cfba3(new BN(MIN_TICK).toTwos(24).toString())
+      const result = await tickMath.MIN_SQRT_RATIO_ee8847ff()
+      expect(min[0].toNumber()).to.eq(result[0].toNumber())
+      expect(min[0].toNumber()).to.eq(MIN_SQRT_RATIO)
+    })
+  })
+
+  describe('#MAX_SQRT_RATIO', async () => {
+    it('equals #getSqrtRatioAtTick(MAX_TICK)', async () => {
+      const max = await tickMath.getSqrtRatioAtTick_986cfba3(new BN(MAX_TICK).toString())
+      const result = await tickMath.MAX_SQRT_RATIO_6d2cc304()
+      expect(max[0].toString()).to.eq(result[0].toString())
+      expect(max[0].toString()).to.eq(MAX_SQRT_RATIO.toString())
+    })
+  })
+
+  describe('#getTickAtSqrtRatio', () => {
+    /*
+    it('throws for too low', async () => {
+      expect(await tickMath.getTickAtSqrtRatio_4f76c058((MIN_SQRT_RATIO.sub(1)).toString())).to.be.revertedWith('R')
+    })
+
+    it('throws for too high', async () => {
+      expect(await tickMath.getTickAtSqrtRatio_4f76c058((MAX_SQRT_RATIO).toString())).to.be.revertedWith('R')
+    })
+    */
+    it('ratio of min tick', async () => {
+      const result = await tickMath.getTickAtSqrtRatio_4f76c058("4295128739")
+      expect(result[0].toString()).to.eq(new BN(MIN_TICK).toTwos(24).toString())
+    })
+    it('ratio of min tick + 1', async () => {
+      const result = await tickMath.getTickAtSqrtRatio_4f76c058('4295343490')
+      expect(result[0].toString()).to.eq(new BN(MIN_TICK + 1).toTwos(24).toString())
+    })
+    it('ratio of max tick - 1', async () => {
+      const result = await tickMath.getTickAtSqrtRatio_4f76c058('42914613736366300043187065181887844931066902546562495343490')
+      expect(result[0].toString()).to.eq(new BN(MAX_TICK - 1).toString())
+    })
+    it('ratio closest to max tick', async () => {
+      const result = await tickMath.getTickAtSqrtRatio_4f76c058((MAX_SQRT_RATIO.sub(1)).toString())
+      expect(result[0].toNumber()).to.eq(MAX_TICK - 1)
+    })
+
+    for (const ratio of [
+      4295128739,
+      encodePriceSqrt(1, (new BN(10).pow(new BN(6))).toString()),
+      encodePriceSqrt(1, (new BN(10).pow(new BN(12))).toString()),
+      encodePriceSqrt(1, 64),
+      encodePriceSqrt(1, 8),
+      encodePriceSqrt(1, 2),
+      encodePriceSqrt(1, 1),
+      encodePriceSqrt(2, 1),
+      encodePriceSqrt(8, 1),
+      encodePriceSqrt(64, 1),
+      encodePriceSqrt((new BN(10).pow(new BN(12))).toString(), 1),
+      encodePriceSqrt((new BN(10).pow(new BN(6))).toString(), 1),
+      MAX_SQRT_RATIO.sub(1),
+    ]) {
+      describe(`ratio ${ratio}`, () => {
+        it('is at most off by 1', async () => {
+          const jsResult = new Decimal(ratio.toString()).div(new Decimal(2).pow(96)).pow(2).log(1.0001).floor()
+          const result = await tickMath.getTickAtSqrtRatio_4f76c058(new BN(ratio.toString()).toTwos(24).toString())
+          const absDiff = new Decimal(result[0].toNumber()).sub(jsResult).abs()
```

Comment:
Since toNumber() is limiting, `result[0].toString()` would be a better choice here.